internal/signerverifier/ssh/ssh.go

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache-2.0
+package ssh
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/base64"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/hiddeco/sshsig"
+	"golang.org/x/crypto/ssh"
+)
+
+// SSH Key/Verifier
+type Key struct {
+	KeyType string
+	KeyVal  KeyVal
+	Scheme  string
+	keyID   string
+}
+
+type KeyVal struct {
+	Public string
+}
+
+// SSH Key.Verify() implementation of dsse.Verifier interface.
+func (k *Key) Verify(_ context.Context, data []byte, sig []byte) error {
+	pub, err := parseSSH2Body(k.KeyVal.Public)
+	if err != nil {
+		return fmt.Errorf("failed to parse ssh public key material: %w", err)
+	}
+
+	signature, err := sshsig.Unarmor(sig)
+	if err != nil {
+		return fmt.Errorf("failed to parse ssh signature: %w", err)
+	}
+
+	message := bytes.NewReader(data)
+
+	// ssh-keygen uses sha512 to sign with **any*** key
+	hash := sshsig.HashSHA512
+	if err = sshsig.Verify(message, signature, pub, hash, "gittuf"); err != nil {
+		return fmt.Errorf("failed to verify ssh signature: %w", err)
+	}
+
+	return nil
+}
+
+// SSH Key.KeyID() implementation of dsse.Verifier interface.
+// FIXME: consider removing error in interface; the dsse implementation
+// clearly needs one
+func (k *Key) KeyID() (string, error) {
+	return k.keyID, nil
+}
+
+// SSH Key.Public() implementation of dsse.Verifier interface.
+// FIXME: consider removing in interface, "Verify()" is all that's needed
+func (k *Key) Public() crypto.PublicKey {
+	sshKey, _ := parseSSH2Body(k.KeyVal.Public)
+	return sshKey.(ssh.CryptoPublicKey).CryptoPublicKey()
+}
+
+// SSH Signer
+type Signer struct {
+	Key  *Key
+	Path string
+}
+
+// SSH Signer.Sign() implementation of dsse.Signer interface.
+// Signs using "s.Path" to a public or private, encrypted or plaintext, rsa,
+// ecdsa or ed25519 key file supported by "ssh-keygen". This aligns with the
+// git "user.signingKey" option.
+// https://git-scm.com/docs/git-config#Documentation/git-config.txt-usersigningKey
+func (s *Signer) Sign(_ context.Context, data []byte) ([]byte, error) {
+	cmd := exec.Command("ssh-keygen", "-Y", "sign", "-n", "gittuf", "-f", s.Path) //nolint:gosec
```

Comment:
Linter flagged this as: "G204: Subprocess launched with a potential tainted input or cmd arguments"

I think it's fine, because we are not running in a shell. Or should we be concerned?

Btw. the linter did not flag the very [similar line below](https://github.com/gittuf/gittuf/pull/414/files#diff-32b875bd523cb3ae662774771a22b3a4ac06ac2170939d1d7738886160c60e04R101).

Reply:
Based on the docs for os/exec:

> Unlike the "system" library call from C and other languages, the os/exec package intentionally does not invoke the system shell and does not expand any glob patterns or handle other expansions, pipelines, or redirections typically done by shells. The package behaves more like C's "exec" family of functions. To expand glob patterns, either call the shell directly, taking care to escape any dangerous input, or use the path/filepath package's Glob function. To expand environment variables, use package os's ExpandEnv. 

I think this is fine, we can `nolint` and move on. @wlynch could you confirm?

Reply:
> Btw. the linter did not flag the very [similar line below](https://github.com/gittuf/gittuf/pull/414/files#diff-32b875bd523cb3ae662774771a22b3a4ac06ac2170939d1d7738886160c60e04R101).

The linter's been hit or miss for me on these...I haven't analyzed what patterns trigger it and what do not with exec.Cmd.